packages/core/src/queries/user.ts

```diff
@@ -97,12 +103,19 @@ export const createUserQueries = (pool: CommonQueryMethods) => {
     );
 
   const hasUser = async (username: string, excludeUserId?: string) =>
-    pool.exists(sql`
-      select ${fields.id}
-      from ${table}
-      where ${fields.username}=${username}
-      ${conditionalSql(excludeUserId, (id) => sql`and ${fields.id}<>${id}`)}
-    `);
+    process.env.CASE_INSENSITIVE_USERNAME === '1'
+      ? pool.exists(sql`
+        select ${fields.id}
+        from ${table}
+        where lower(${fields.username})=lower(${username})
```

Comment:
I think this line can be a conditional sql

Reply:
I *think* I figured out how to do it with `conditionalSql`while keeping it readable, let me know if there is a better way!